src/Agent/NewRelic/Agent/Core/Attributes/LogContextDataFilter.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Copyright 2020 New Relic, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using NewRelic.Agent.Configuration;
+
+namespace NewRelic.Agent.Core.Attributes
+{
+    public interface ILogContextDataFilter
+    {
+        Dictionary<string, object> FilterLogContextData(Dictionary<string, object> contextData);
+    }
+
+    public class LogContextDataFilter : ILogContextDataFilter
```

Comment:
This filter needs its own test harness to validate all conditions and use cases. This will (hopefully) prevent regressions if anyone attempts to refactor this in the future.

Reply:
Done.